ocaml/sdk-gen/go/gen_go_helper.ml

```diff
@@ -325,40 +388,57 @@ module Json = struct
 
   let messages_of_obj obj =
     let ctor_fields = ctor_fields_of_obj obj in
-    let params_in_msg msg =
-      if msg.msg_session then
-        session_id :: msg.msg_params
-      else
-        msg.msg_params
+    let method_name_exported msg params latest =
+      let method_name = snake_to_camel msg.msg_name in
+      let len = List.length params in
+      let num =
+        match (String.lowercase_ascii msg.msg_obj_name, msg.msg_session) with
+        | "session", true when len > 1 ->
+            len - 1
+        | _ ->
+            len
+      in
+      if latest then method_name else method_name ^ string_of_int num
     in
-    List.map
-      (fun msg ->
-        let params = params_in_msg msg |> of_params in
-        let base_assoc_list =
-          [
-            ("method_name", `String msg.msg_name)
-          ; ("class_name", `String obj.name)
-          ; ("class_name_exported", `String (snake_to_camel obj.name))
-          ; ("method_name_exported", `String (snake_to_camel msg.msg_name))
-          ; ("description", desc_of_msg msg ctor_fields)
-          ; ("result", of_result obj msg)
-          ; ("params", params)
-          ; ("errors", of_errors msg.msg_errors)
-          ; ("has_error", `Bool (msg.msg_errors <> []))
-          ; ("async", `Bool msg.msg_async)
-          ]
-        in
-        (* Since the param of `session *Session` isn't needed in functions of session object,
-           we add a special "func_params" field for session object to ignore `session *Session`.*)
-        if obj.name = "session" then
-          `O
-            (("func_params", msg.msg_params |> of_params)
-            :: (add_session_info obj.name msg.msg_name @ base_assoc_list)
-            )
-        else
-          `O base_assoc_list
-      )
-      obj.messages
+    obj.messages
+    |> List.rev_map (fun msg ->
+           let of_message (latest, params, rel_version) =
+             let base_assoc_list =
+               [
+                 ("method_name", `String msg.msg_name)
+               ; ("class_name", `String obj.name)
+               ; ("class_name_exported", `String (snake_to_camel obj.name))
+               ; ( "method_name_exported"
+                 , `String (method_name_exported msg params latest)
```

Comment:
The matching on `obj.name, msg.msg_session` in this function can be shared with the one in `of_message`.